zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -3613,6 +3613,45 @@ paths:
                 $ref: "#/components/schemas/ProblemDetail"
         "500":
           $ref: "#/components/responses/InternalServerError"
+    patch:
+      tags:
+        - User
+      operationId: updateUser
+      summary: Update user
+      description: |
+        Updates a user.
+      parameters:
+        - name: username
+          in: path
+          required: true
+          description: The username of the user to update.
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/UserUpdateRequest"
+        required: true
+      responses:
+        "204":
```

Comment:
❓ WDYT about returning the user object as a response here as well?

Reply:
👍 We should, it's defined in the [internal definition doc](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit?pli=1&tab=t.0#heading=h.hxbk2uscj4u0). Other update endpoints follow this pattern as well.

Reply:
@remcowesterhoud I had another look, and most "update" endpoints that we currently have don't return a response. Should we align with the REST API team before we make this change?

Reply:
Yes please check in with them. I'm fine with not doing it if that's the result. It's something we can always add later on.

Reply:
Results from the REST API team discussion [here](https://camunda.slack.com/archives/C06UKS51QV9/p1736953822611169) (internal link). The general feeling is that our idea hasn't been explored thoroughly, so if we make this change:
- we would break UX consistency of the REST API (update) endpoints.
- we would make the eventual consistency between the (Zeebe) runtime and secondary storage more visible, which is bad UX.

I will instead create a `feature` issue where this approach can be explored in more detail.

Reply:
Thanks for raising it @koevskinikola 👍 In that case this is good to merge 🚀